traits/trait_base.py

```diff
@@ -33,6 +34,8 @@
 
 SequenceTypes = (list, tuple)
 
+EnumTypes = (list, tuple, set, enum.EnumMeta)
+
```

Comment:
Supporting `set` here is new, right? I'm not convinced we want to do that: in particular, getting unpredictable defaults could be a problem.

Reply:
Ah - I had assumed that set had picked up the ordering behaviour of dicts.  I will remove this.